analysers/Analyser_Osmosis.py

```diff
@@ -355,14 +355,31 @@ def dump_delete(self):
             types = {'N': 'node', 'W': 'way', 'R': 'relation'}
             for t, ids in self.already_issued_objects.items():
                 if ids:
-                    sql = "SELECT v.id, l.id FROM (VALUES ({0})) AS v(id) LEFT JOIN {1}s AS l ON v.id = l.id WHERE l.id IS NULL;".format('),('.join(map(str, ids)), types[t])
+                    sql = """
+-- Touched
+SELECT id
+FROM touched_{1}s
+    JOIN (VALUES ({0})) AS v(id) USING (id)
+UNION ALL
+-- Deleted
+SELECT id
+FROM (VALUES ({0})) AS v(id)
+    LEFT JOIN {1}s AS l USING(id)
+WHERE l.id IS NULL
+"""
+                    sql = sql.format('),('.join(map(str, ids)), types[t])
                     self.giscurs.execute(sql)
                     for res in self.giscurs.fetchall():
                         self.error_file.delete(types[t], res[0])
         else:
             # Change
             for t in ["node", "way", "relation"]:
-                sql = "(SELECT id FROM actions WHERE data_type='{0}' AND action='D') UNION ALL (SELECT id FROM touched_{1}s) EXCEPT (SELECT id FROM actions WHERE data_type='{0}' AND action='C')".format(t[0].upper(), t)
+                sql = """
+SELECT id FROM actions WHERE data_type='{0}' AND action IN ('D', 'C')
```

Comment:
Does it make a difference to add 'C'?

Created nodes/ways/relations shouldn't already be present in osmose database - I'm not sure that osm ids can be re-used for a new object.

In any case, I'm fine with this change.

Reply:
This part of the code if for change mode. Is not used.
But recreated object may also change it at the same time.
The new 'C' come with the remove of the  `EXCEPT` part.